src/main/kotlin/uk/gov/justice/probation/courtcaseservice/service/CaseWorkflowService.kt

```diff
@@ -24,6 +23,19 @@ class CaseWorkflowService(val hearingRepository: HearingRepository, val courtRep
             })
     }
 
+    fun updateHearingOutcome(hearingId: String, hearingOutcomeState: String, assignedTo: String, assignedToUuid: String) {
+        hearingRepository.findFirstByHearingId(hearingId).ifPresentOrElse(
+                { hearingEntity: HearingEntity ->
+                    hearingEntity.hearingOutcome.updateState(hearingOutcomeState)
```

Comment:
If you want to mutate a single field setter method is the standard, use the @Setter annotation on that field and you could do hearingEntity.hearingOutcome.state = hearingOutcomeState
same for all 3 fields

Reply:
This method can be tailored to assign flow rather than generalised update and avoid controller needing to worry about what state it is going to be .. this is service's responsibility 

call service method "fun assignOutcomeTo(user, uuid)' and don't have to pass outcome state which is implied here

If you consider -> instead of having 3 field mutators below is more domain oriented way.

HearingOutcomeEntity.assignTo(user, uuid) {
  // update all 3 fields here 
  // the state IN_PROGRESS is implied here, no need to pass it around all the way from the controller just set it.
}

Reply:
Does this endpoint also responsible for updating the final status of the outcome?  @rschallapalli @johnyfarrar